src/main/java/net/earthcomputer/clientcommands/command/AliasCommand.java

```
@@ -0,0 +1,174 @@
+            .then(literal("remove")
+                .then(argument("key", string())
+                    .executes(ctx -> removeAlias(getString(ctx, "key"))))));
+
+        HashMap<String, String> aliasMap = getAliases();
+
+        if (aliasMap == null || aliasMap.isEmpty()) {
+            sendError(new LiteralText("No aliases registered"));
+        } else {
+            for(String key: aliasMap.keySet()) {
+                addClientSideCommand(key);
+                dispatcher.register(literal(key)
+                        .executes(ctx -> executeAliasCommand(key, null))
+                        .then(argument("arguments", greedyString())
+                                .executes(ctx -> executeAliasCommand(key, getString(ctx, "arguments")))));
+            }
+        }
+
+    }
+    private static int executeAliasCommand(String aliasKey, String arguments) {
+
+        // TODO: add support for optional greedy text arguments(?)
+        HashMap<String, String> aliasMap = getAliases();
+        String cmd;
+        if(aliasMap.containsKey(aliasKey)) {
+            cmd = aliasMap.get(aliasKey);
+        } else {
+            sendError(new LiteralText("No such key exists"));
+            return 0;
+        }
+
+        if(arguments!=null) {
+            cmd = String.format(cmd, arguments.split(" "));
+        }
+        MinecraftClient.getInstance().player.sendChatMessage(cmd);
+
+        return 0;
+    }
+
+    private static int addAlias(String key, String command) {
+        ClientPlayNetworkHandler networkHandler = MinecraftClient.getInstance().getNetworkHandler();
+        assert networkHandler != null;
+        var dispatcher = (CommandDispatcher<ServerCommandSource>) (CommandDispatcher<?>) networkHandler.getCommandDispatcher();
+
+        HashMap<String, String> aliasMap = getAliases();
+
+        assert aliasMap != null;
+        if(aliasMap.containsKey(key)) {
+            sendError(new LiteralText("That key already exists!"));
```

You should probably throw an exception rather than sending an error message.

---

Since this is user feedback, that an alias already exists, I don't think an exception should be thrown.
And I also think, that if an alias already exists the old alias should be overwritten.

---

I like the suggestion of overwriting the existing alias, perhaps with a clickable [undo?] button in chat if the alias was overwritten by accident

---

Following the style of `ckit` and `citemgroup`, an `ALREADY_EXISTS_EXCEPTION` should be thrown.